### ISSUE TITLE
Configure pre-commit hooks

### ISSUE THREAD

**Issue by chshersh** [chshersh/iris#69]
Currently, only for whitespace cleanups. Later, things like `hlint` and `fourmolu` will be added.

Examples can be found here:

* https://github.com/chshersh/tool-sync/blob/main/.pre-commit-config.yaml
* https://github.com/chshersh/tool-sync#pre-commit-hooks

**Comment by aleeusgr** [comment]
I started work on this.
As far as I understand, pre-commit hooks trigger `git` upon commit; therefore, to test I need to first commit `.pre-commit-config.yaml` to my fork, and then test it's functionality by pushing new commits. 

Could you, please, describe the desired effect? I don't understand what is meant by whitespace cleanups.
Thank you

PS. I've seen `hlint` and `fourmolu` pre-commit hooks in another library I contributed to, maybe it'll be easier then to add their functionality first?

Thank you

**Comment by aleeusgr** [comment]
i've met this idea [here](https://github.com/tweag/monad-bayes/blob/e18f4f37157dc966c104df00a384391d002c3ede/flake.nix#L18)
however this repo does not use `nix`, so it would seem pre-commit is a tool I have to have locally in order for me to be able to test how the config file works

**Comment by aleeusgr** [comment]
reading [this guide](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)
The hooks are all stored in the hooks subdirectory of the Git directory. In most projects, that’s `.git/hooks`.
The pre-commit hook is run first, before you even type in a commit message. It’s used to inspect the snapshot that’s about to be committed.

[This guide](https://pre-commit.com/)

 ou can generate a very basic configuration using [pre-commit sample-config](https://pre-commit.com/#pre-commit-sample-config)

**Comment by aleeusgr** [comment]
`$ pre-commit sample-config > .pre-commit-config.yaml`
'$ pre-commit run`
```
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check Yaml...........................................(no files to check)Skipped
Check for added large files..............................................Passed
```

**Comment by aleeusgr** [comment]
https://github.com/chshersh/iris/pull/72

**Comment by aleeusgr** [comment]
[here](https://github.com/bwbaugh/haskell-pre-commit-hooks/blob/master/.pre-commit-hooks.yaml) are a hook for hlint and styish-haskell
and [fourmolu](https://github.com/fourmolu/fourmolu/blob/main/.pre-commit-config.yaml)

**Comment by chshersh** [comment]
Hey @aleeusgr, thanks for the contribution!

Adding `hlint` would be nice indeed, so feel free to add it after 🙂 
I haven't looked intro `fourmolu` and haven't created the config that satisfies my taste so no rush in adding formatter here 👍🏻

**Comment by chshersh** [comment]
Closed in #72
